src/xhr.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assert} from './asserts';
+
+
+/**
+ * The "init" argument of the Fetch API. Currently, only "credentials: include"
+ * is implemented.
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
+ *
+ * @typedef {{
+ *   credentials: string
+ * }}
+ */
+const FetchInit = {};
+
+
+/**
+ * Fetches and constructs JSON object based on the fetch polyfill.
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
+ *
+ * @param {string} input
+ * @param {?FetchInit=} opt_init
+ * @return {!Promise<!JSON>}
+ */
+export function fetchJson(input, opt_init) {
+  return fetch(input, opt_init).then(response => {
+    return response.json();
+  });
+}
+
+
+/**
+ * A minimal polyfill of Fetch API. It only polyfills what we currently use.
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
+ *
+ * Notice that the "fetch" method itself is not exported as that would require
+ * us to immediately support a much wide API.
+ *
+ * @param {string} input
+ * @param {?FetchInit=} opt_init
+ * @return {!Promise<!FetchResponse>}
+ */
+function fetch(input, opt_init) {
+  assert(typeof input == 'string', 'Only URL supported: %s', input);
+  const init = opt_init || {};
+  assert(!init.credentials || init.credentials == 'include',
+      'Only credentials=include support: %s', init.credentials);
+
+  return new Promise(function(resolve, reject) {
+    const xhr = createXhrRequest(init.method || 'GET', input);
```

Comment:
Hmm. Good point. We can certainly use fetch in this case. I'll prioritize `window.fetch` when available.